### PR TITLE
Adding github action for automated submission and dependabot upkeep

### PR DESCRIPTION
**Pull request by louisgv**
Hi @roedesh!

Love copy guard! I have been spreading it to my friends who're into cyber security :D I was looking for ways to contribute, and found that since your codebase doesn't have automated ci, I thought adding that might be helpful. 

First, I added a dependabot config for you so it should bump dependencies version automatically. Then, I created 2 workflows in this PR:

- `automerge`: automate the dependabot merging to reduce PR noise, using https://github.com/fastify/github-action-merge-dependabot
- `submit`: automate the zip submission process to the Chrome/Firefox/Edge stores automatically, using [bpp](https://browser.market) . It is set to run manually from the github action tab, but we can tweak it to run as frequent as you wish! The only thing you would need to create is a `SUBMIT_KEYS` github repository secret.

The `SUBMIT_KEYS` secret is a json, with the schema defined [here](https://github.com/plasmo-corp/bpp/blob/v1/keys.schema.json).

Here's a sample key for you:

```json
{
  "$schema": "https://raw.githubusercontent.com/plasmo-corp/bpp/v1/keys.schema.json",
  "chrome": {
    "clientId": "123",
    "refreshToken": "789",
    "extId": "abcd"
  },
	"firefox": {
		"apiKey": "123",
		"apiSecret": "789",
		"extId": "abcd"
	},
	"edge": {
		"cookie": "123",
		"extId": "abcd"
	}
}

```

You can find instructions on how to get those keys in the schema, or if you use vscode, the schema should provide hint/intelisense when hovering over the json properties. If you need any help in setting up the keys, feel free to @ me! Otherwise if this doesn't seem necessary, feel free to close the PR :)

### CODE REVIEW

**Comment by roedesh** [comment]
Hi @louisgv 

Thank you for the contribution! Automated CI was on my todo-list, so I really appreciate the effort. I'm glad that you and your friends like the extension.

I have some questions/remarks:

- If I understand correctly, the "Submit" workflow will only run when I manually trigger it. Can this also be triggered when I push a new tag?
- For the Dependabot workflow, I'd like to run the unit tests before the PR is automatically merged, to make sure everything keeps working. I suppose this could be a separate workflow, that needs to succeed before the PR can be merged.

Also, the Submit workflow will only submit the zip file, meaning I can still manually fill in the other details before I publish the new version. Is that correct?

Great work so far!

**Comment by louisgv** [comment]
> I have one remark. For the Dependabot workflow, I'd like to run the unit tests before the PR is automatically merged, to make sure everything keeps working. I suppose this could be a separate workflow, that needs to succeed before the PR can be merged.

What you described is precisely what the automerge workflow is doing: https://github.com/roedesh/copyguard/pull/5/files#diff-cc8c65daed8907e6bb50ac1769d49c05f5f48bdbe8b5cfd3b24b7c5e56ceb8dcR6-R15

The `automege` step requires the `build` step to be finished and succeed before it would run. We can rename `build` to `test`, and add `yarn test` to that job :D

> Also, the Submit workflow will only submit the zip file, meaning I can still manually fill in the other details before I publish the new version. Is that correct?

Yup, this is precisely the case for this workflow!

File: .github/workflows/automerge.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: automerge
+on: [push, pull_request]
+
+jobs:
+  build:
```
Comment (roedesh) [inline .github/workflows/automerge.yml:5]: Could you add another job here for the unit tests called `test`, and make `automerge` require it? Or perhaps you can add a extra step in the `build` job. I'm not sure which of the two is better, so I'll leave that up to you.

If you do that, then I will merge this PR :)

Comment (louisgv) [reply .github/workflows/automerge.yml]: Done :D - I added `yarn test` right after build, which means if either `yarn build` or `yarn test` fail, the `automerge` step won't be triggered.

Comment (roedesh) [reply .github/workflows/automerge.yml]: Thanks you!

**Pull request merged**